### PR TITLE
Ft add study groups

### LINKED ISSUE #125: Add Possibility to add new study groups
Add Command to be able to add new Study groups.

Need to be careful to remove all instances in the code where the 3 current study groups are fixed.

### PR DESCRIPTION
**Pull request by marcelropos**
Add study groups and store color and activation status.

Closes #125

### CODE REVIEW

File: cogs/study_groups.py [hunk 1]
```diff
@@ -134,6 +134,43 @@ async def _group(self, ctx: Context):
         member: Union[Member, User] = ctx.author
         logger.info(f'User="{member.name}#{member.discriminator}({member.id})", Command="{ctx.message.content}"')
 
+    @_group.command(pass_context=True,
+                    name="create",
+                    brief="Creates a Study group.",
+                    help="The name must contain the tag and the semester number.")
+    async def group_create(self, ctx: Context, name: str, semester: int = 1, color: int = 0):
```
Comment (maxwai) [inline cogs/study_groups.py:141]: make color a needed parameter with no default, force the User to give one.

File: cogs/study_groups.py [hunk 2]
```diff
@@ -134,6 +134,43 @@ async def _group(self, ctx: Context):
...
+            color: color of group.
+
+            semester: number of semester
+        """
+        color_db = PrimitiveMongoData(CollectionEnum.GROUP_COLOR)
+        active_db = PrimitiveMongoData(CollectionEnum.GROUP_ACTIVE)
+        guild: Guild = ctx.guild
+
+        await color_db.insert_one({name: color})
+        await active_db.insert_one({name: True})
```
Comment (maxwai) [inline cogs/study_groups.py:159]: check if there isn't already a group with that name

File: cogs/study_groups.py [hunk 3]
```diff
@@ -134,6 +134,43 @@ async def _group(self, ctx: Context):
...
+                (
+                    await StudySubjectUtil.get_server_objects(
+                        ConfigurationNameEnum.GROUP_CATEGORY,
+                        guild,
+                        f"{name}{i}",
+                        ConfigurationNameEnum.STUDY_SEPARATOR_ROLE,
+                        self.db,
+                        color=Color(color)
+                        , reason="Create new group")
+                ).role)
```
Comment (maxwai) [inline cogs/study_groups.py:172]: give some kind of feedback to the user that the groups are created

**Review by maxwai — CHANGES_REQUESTED** [review]

**Comment by maxwai** [comment]
Also fixes part of #126

File: cogs/study_groups.py (hunk 1 above)
Comment (marcelropos) [reply cogs/study_groups.py]: Okay is required now

File: cogs/study_groups.py (hunk 2 above)
Comment (marcelropos) [reply cogs/study_groups.py]: Will be checked new

File: cogs/study_groups.py (hunk 3 above)
Comment (marcelropos) [reply cogs/study_groups.py]: Feedback for every group creation as reply.

File: cogs/study_groups.py [hunk 4]
```diff
@@ -155,21 +155,25 @@ async def group_create(self, ctx: Context, name: str, semester: int = 1, color:
         active_db = PrimitiveMongoData(CollectionEnum.GROUP_ACTIVE)
         guild: Guild = ctx.guild
 
-        await color_db.insert_one({name: color})
-        await active_db.insert_one({name: True})
+        if not await color_db.find_one({name: {"$exists": True}}):
+            await color_db.insert_one({name: color})
+        if not active_db.find_one({name: {"$exists": True}}):
+            await active_db.insert_one({name: True})
```
Comment (maxwai) [inline cogs/study_groups.py:161]: you only need to check one database since they are mirrors from each other

Comment (marcelropos) [reply cogs/study_groups.py:161]: Rejected.
Each entry is independent of another and must be checked individually.
It does not matter that they are created together.

File: cogs/study_groups.py [hunk 5]
```diff
@@ -134,6 +134,47 @@ async def _group(self, ctx: Context):
...
+
+        for i in range(1, semester + 1):
+            created = await StudySubjectUtil.get_server_objects(
+                ConfigurationNameEnum.GROUP_CATEGORY,
+                guild,
+                f"{name}{i}",
+                ConfigurationNameEnum.STUDY_SEPARATOR_ROLE,
+                self.db,
+                color=Color(color)
```
Comment (maxwai) [inline cogs/study_groups.py:170]: check if color is an int, if it is a str, try to convert hex to decimal

**Review by maxwai — CHANGES_REQUESTED** [review]

File: cogs/study_groups.py (hunk 5 above)
Comment (maxwai) [reply cogs/study_groups.py:170]: To be more specific, do the check before inserting into database

Comment (marcelropos) [reply cogs/study_groups.py:170]: Not necessary type conversion already done by function definition and framework magic.

Comment (maxwai) [reply cogs/study_groups.py:170]: no, not the case, see screenshots:

image

image

Comment (marcelropos) [reply cogs/study_groups.py:171]: You're right. Fixed

**Review by maxwai — APPROVED** [review]

**Pull request merged**